src/java/src/test/java/ai/onnxruntime/genai/GenerationTest.java

```diff
@@ -25,79 +25,132 @@ public class GenerationTest {
   // phi-2 can be used in full end-to-end testing but needs to be manually downloaded.
   // it's also used this way in the C# unit tests.
   private static final String phi2ModelPath() {
-    String repoRoot = TestUtils.getRepoRoot();
-    File f = new File(repoRoot + "examples/python/example-models/phi2-int4-cpu");
-
-    if (!f.exists()) {
-      logger.warning("phi2 model not found at: " + f.getPath());
-      logger.warning(
-          "Please install as per https://github.com/microsoft/onnxruntime-genai/tree/rel-0.2.0/examples/csharp/HelloPhi2");
-      return null;
-    }
-
-    return f.getPath();
+    return TestUtils.getFilePathFromResource("/phi-2/int4/cpu");
   }
 
   @SuppressWarnings("unused") // Used in EnabledIf
   private static boolean havePhi2() {
     return phi2ModelPath() != null;
   }
 
+  @SuppressWarnings("unused") // Used in EnabledIf
+  private static boolean haveAdapters() {
+    return TestUtils.testAdapterTestModelPath() != null;
+  }
+
   @Test
   @EnabledIf("havePhi2")
   public void testUsageNoListener() throws GenAIException {
-    SimpleGenAI generator = new SimpleGenAI(phi2ModelPath());
-    GeneratorParams params = generator.createGeneratorParams("What's 6 times 7?");
-
-    String result = generator.generate(params, null);
-    logger.info("Result: " + result);
-    assertTrue(result.indexOf("Answer: 42") != -1);
+    try (SimpleGenAI generator = new SimpleGenAI(phi2ModelPath());
+        GeneratorParams params =
+            generator.createGeneratorParams(
+                TestUtils.applyPhi2ChatTemplate("What's 6 times 7?")); ) {
+      params.setSearchOption("max_length", 20);
+
+      String result = generator.generate(params, null);
+      logger.info("Result: " + result);
+      assertNotNull(result);
+    }
   }
 
   @Test
   @EnabledIf("havePhi2")
   public void testUsageWithListener() throws GenAIException {
-    SimpleGenAI generator = new SimpleGenAI(phi2ModelPath());
-    GeneratorParams params = generator.createGeneratorParams("What's 6 times 7?");
-    Consumer<String> listener = token -> logger.info("onTokenGenerate: " + token);
-    String result = generator.generate(params, listener);
+    try (SimpleGenAI generator = new SimpleGenAI(phi2ModelPath());
+        GeneratorParams params =
+            generator.createGeneratorParams(
+                TestUtils.applyPhi2ChatTemplate("What's 6 times 7?")); ) {
+      params.setSearchOption("max_length", 20);
 
-    logger.info("Result: " + result);
-    assertTrue(result.indexOf("Answer: 42") != -1);
+      Consumer<String> listener = token -> logger.info("onTokenGenerate: " + token);
+      String result = generator.generate(params, listener);
+
+      logger.info("Result: " + result);
+      assertNotNull(result);
+    }
+  }
+
+  @Test
+  @EnabledIf("haveAdapters")
+  public void testUsageWithAdapters() throws GenAIException {
```

Comment:
Would be great if the test had before/after responses to demonstrate the adapters changing the output. Testing that you get a response with a shape is good, but it won't catch an issue where the response is invalid.

Reply:
Not quite following. Based on the code we can load and set the adapters and get the output. What interfaces are missing to be able to check the response is valid?

Reply:
`OrtValue` itself and related functions like `AllocatorFree` are not exposed by GenAI header, which means a lot of things in https://github.com/microsoft/onnxruntime/blob/main/java/src/main/native/ai_onnxruntime_OnnxTensor.c  cannot be done easily in the GenAI JNI wrapper code.

The current implementation of `GetData` is not complete. For example, the current `GetData` interface lacks the ability to fetch float16 type of data with proper size calculation. I was also bothered by this in #1084.

Reply:
Currently, only limited subset of `GetTensorTypeAndShapeInfo()` is exposed by GenAI header like this:

https://github.com/microsoft/onnxruntime-genai/blob/c5745fd6d91a2ba2d94c6a52f6255652dde02508/src/ort_genai_c.cpp#L477

I believe at least I need to add `GetTensorTypeAndShapeInfo().GetDimensionsCount()` and `GetTensorTypeAndShapeInfo().GetElementCount()` to copy the logic here for proper size calculation:

https://github.com/microsoft/onnxruntime/blob/1128882bfd2a97c20f8a2a5ddb26cb0d42d9ebba/java/src/main/native/OrtJniUtil.c#L257

Reply:
I'll resolve this for now and try to seek a better solution in the future.